### ISSUE TITLE
Argument parsing

### ISSUE THREAD

**Issue by joshperry** [Unitech/PM2#942]
Seems like every time I update PM2 something explodes. This time it's argument parsing, I got to the point where I just had to start adding logging entries in the code to see what was going on.

Here is my process.json file:
```
{
	"apps" : [
		{
			"name": "hubot-tron",
			"script": "./node_modules/.bin/hubot",
			"args": "['--adapter', 'hipchat', '--name', 'tron']",
			"watch": "true",
			"exec_mode": "fork_mode",
			"run_as_user": "app",
			"exec_interpreter": "coffee",
			"watch": false,
			"env": {
				"PORT":"8080",
				"HUBOT_HIPCHAT_JID":"148274_1070946@chat.hipchat.com",
				"HUBOT_HIPCHAT_PASSWORD":"Oh shit, passwords are bad..."
				"HUBOT_HIPCHAT_ROOMS":"148274_dev@conf.hipchat.com",
				"HUBOT_LOG_LEVEL":"debug",
				"HUBOT_AUTH_ADMIN":"Joshua Perry"
			}
		}
	]
}
```

And here is what PM2 sends to `spawn`:
`spawn('coffee /opt/pdk/pdktron/node_modules/.bin/hubot,[,--adapter,,,hipchat,,,--name,,,tron,]', { args: [ '[', '--adapter', ',', 'hipchat', ',', '--name', ',', 'tron', ']' ], { ...snip... });`

**Comment by Tjatse** [comment]
@joshperry The Config part was reconstructed since v0.12.2. try:
```
"args": ["--adapter", "hipchat", "--name", "tron"]
```

or
```
"args": "--adapter hipchat --name tron"
```

Sorry for the confusion.

**Comment by Tjatse** [comment]
References:
- [Schema](https://github.com/Unitech/PM2/blob/development/ADVANCED_README.md#schema)
- [JSON Declaration](https://github.com/Unitech/PM2/blob/development/ADVANCED_README.md#json-app-declaration)

**Comment by joshperry** [comment]
Do breaking changes like this get deprecation warnings? Any kind of thought as to how it will effect existing users? Notices in big red letters in release notes saying "These things _will_ break if you're using them and upgrade!"?

You guys are on the front page of npmjs.org and had almost 53,000 downloads last month, for God's sake!!! This isn't a package that gets installed as a dependency to other libraries, there were 53,000 people that consciously said "Hey I am going to install this version of PM2 on my server to manage processes". Like it or not, you (the project) now have a responsibility to your users to maintain compatibility and deprecate breaking changes in a sane and manageable way.

If you guys don't get your shit together and start putting some rigor into your releases, people are going to stop using the project because of the pain. Expecting your users to read one-line changelog entries, look up the referenced PRs, and read through the code to see if something is going to break when they upgrade is _not_ an effective release management strategy.

This isn't a joke, this is actually what I've had to do _every_ time I've upgraded PM2, I've wasted many cumulative hours hunting down the code changes that broke my install.
